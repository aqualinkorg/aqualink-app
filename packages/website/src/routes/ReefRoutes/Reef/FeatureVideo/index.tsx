import React from "react";
import {
  withStyles,
  WithStyles,
  createStyles,
  Card,
  CardContent,
  Grid,
  Typography,
  IconButton,
  Theme,
} from "@material-ui/core";
import PlayArrowIcon from "@material-ui/icons/PlayArrow";
import ReactPlayer from "react-player";

import reefImage from "../../../../assets/reef-image.jpg";
import uploadIcon from "../../../../assets/icon_upload.svg";

const FeatureVideo = ({ url, classes }: FeatureVideoProps) => {
  return (
    <Card className={classes.card}>
      {url ? (
        <CardContent className={classes.content}>
          <ReactPlayer
            height="100%"
            width="100%"
            playing
            muted
            light
            url={`${url}`}
          />
        </CardContent>
      ) : (
        <div className={classes.noVideoCard}>
          <div className={classes.noVideoCardHeader}>
            <Grid container item spacing={2}>
              <Grid container justify="center" item xs={12}>
                <Typography style={{ opacity: 0.5 }} variant="h5">
                  VIDEO TO BE UPLOADED
                </Typography>
              </Grid>
              <Grid container justify="center" item xs={12}>
                <IconButton>
                  <img src={uploadIcon} alt="upload" />
                </IconButton>
              </Grid>
            </Grid>
          </div>
          <div className={classes.noVideoCardContent} />
          <IconButton className={classes.playIconWrapper}>
            <PlayArrowIcon className={classes.playIcon} />
          </IconButton>
        </div>
      )}
    </Card>
  );
};

const styles = (theme: Theme) => {
  return createStyles({
    card: {
      height: "100%",
      width: "100%",
      display: "flex",
    },
    content: {
      height: "100%",
      width: "100%",
      padding: "0",
    },
    noVideoCard: {
      position: "relative",
      height: "100%",
      width: "100%",
    },
    noVideoCardHeader: {
      backgroundColor: "#033042",
      opacity: 0.75,
      position: "absolute",
      top: 0,
      width: "100%",
      padding: "2rem 0 1rem",
      zIndex: 1,
    },
    noVideoCardContent: {
      width: "100%",
      height: "100%",
      backgroundImage: `url(${reefImage})`,
      backgroundRepeat: "no-repeat",
      backgroundSize: "cover",
      filter: "blur(2px)",
    },
    playIconWrapper: {
      position: "absolute",
      top: "50%",
      left: "50%",
      transform: "translate(-50%, 0)",
    },
    playIcon: {
      opacity: 0.5,
      color: theme.palette.primary.light,
      fontSize: "8rem",
    },
  });
};

interface FeatureVideoIncomingProps {
  url: string | null;
}

type FeatureVideoProps = WithStyles<typeof styles> & FeatureVideoIncomingProps;

export default withStyles(styles)(FeatureVideo);
