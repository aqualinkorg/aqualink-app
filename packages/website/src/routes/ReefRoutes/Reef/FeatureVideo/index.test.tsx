import React from "react";
import { render } from "@testing-library/react";

import FeatureVideo from ".";

test("renders as expected", () => {
  const { container } = render(<FeatureVideo url={null} />);
  expect(container).toMatchSnapshot();
});
