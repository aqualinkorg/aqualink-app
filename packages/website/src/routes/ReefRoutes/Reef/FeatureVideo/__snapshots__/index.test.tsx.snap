// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<div>
  <div
    class="MuiPaper-root MuiCard-root FeatureVideo-card-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="FeatureVideo-noVideoCard-3"
    >
      <div
        class="FeatureVideo-noVideoCardHeader-4"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item"
        >
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-xs-center MuiGrid-grid-xs-12"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5"
              style="opacity: 0.5;"
            >
              VIDEO TO BE UPLOADED
            </h5>
          </div>
          <div
            class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-justify-xs-center MuiGrid-grid-xs-12"
          >
            <button
              class="MuiButtonBase-root MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <img
                  alt="upload"
                  src="icon_upload.svg"
                />
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="FeatureVideo-noVideoCardContent-5"
      />
      <button
        class="MuiButtonBase-root MuiIconButton-root FeatureVideo-playIconWrapper-6"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root FeatureVideo-playIcon-7"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M8 5v14l11-7z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
